### PR TITLE
Validates health Id in user auth flow

### PR DESCRIPTION
**Pull request by MeghnaThoughtworks**

### CODE REVIEW

**Review by praveenadayanand — APPROVED** [review]

**Review by swatigogia2020 — APPROVED** [review]

**Comment by praveenadayanand** [comment]
@MeghnaThoughtworks unittests are failing

**Review by praveenadayanand — CHANGES_REQUESTED** [review]
Unittest cases need to be fixed

**Review by praveenadayanand — APPROVED** [review]
Just for my understanding, is getting cmsuffix as part of the incoming health-id instead of a configuration, part of this story card ?

**Comment by MeghnaThoughtworks** [comment]
> Just for my understanding, is getting cmsuffix as part of the incoming health-id instead of a configuration, part of this story card ?

I made the cmSuffix come from the config and not from the health Id as part of this card. So the opposite of what you said

File: src/In.ProjectEKA.HipService/DataFlow/ErrorMessage.cs [hunk 1]
```diff
@@ -8,5 +8,8 @@ public static class ErrorMessage
         public static readonly string HealthInformationNotFound = "Health information not found";
         public static readonly string LinkExpired = "Link has expired";
         public static readonly string ExpiredKeyPair = "Key material expired";
+        public static readonly string InvalidHealthId = "Health Id is invalid. Must contain at least 4 letters. " +
+                                                        " We only allow alphabets and numbers and" +
+                                                        " do not allow special character except dot (.)";
```
Comment (Mahesh-I) [inline src/In.ProjectEKA.HipService/DataFlow/ErrorMessage.cs:13]: Do we really need error messages like these, usually these will be shown while creating a new health id right? Health Id is invalid or wrong  is enough i think

**Pull request merged**